### ISSUE TITLE
Feature: image_dataset_from_directory support one_hot/categorical labels

### ISSUE THREAD

**Issue by skyetomez** [keras-team/autokeras#1872]
### Code Example
```python
ak.keras.utils.image_dataset_from_directory(
    directory,
    labels='inferred',
    label_mode='int',
    class_names=None,
    color_mode='rgb',
    batch_size=32,
    image_size=(256, 256),
    shuffle=True,
    seed=None,
    validation_split=None,
    subset=None,
    interpolation='bilinear',
    follow_links=False,
    crop_to_aspect_ratio=False,
    **kwargs
)

```

### Reason
This would make using he ak generator of image datasets from a directory more convenient
